### PR TITLE
feat: 🍰 Allow Only Supported Image File Formats 

### LINKED ISSUE #3923: 🐛 [Bug] post gets created even if the user uploads unsupported file formats
while creating post if the user uploads image of format other than supported ( jpg/png/gif) the post gets created and the post appears as : 

Screenshot from 2020-10-26 01-00-53

### Steps to reproduce the behavior
1. create  a  new post 
2. upload a image of file format other that ( jpg/png/gif)
3. click the confirm button 
4. now check the created post 

### Expected behavior

the post should not get created or warn ( or just inform the user ) about the supported file formats

### Version & Environment
 Type: [ Desktop ]      
 - OS: [ubuntu ]      
 - Browser: [ chrome 85.0.4183.83 ] 
 - Version [ ]  
 - Device: []  

### Additional context

 I think this issue can be resolved if user is restricted to upload only supported file formats , like .. when the user clicks the upload button only the supported file formats are filtered from users device ( as we do it in html by ```accept="image/*"```  for the file input field)

### PR DESCRIPTION
**Pull request by dasari810**
## 🍰 Pullrequest

- to automatically  filter only the supported image file formats when the user uploads the image for creating post 

- to give an error message if an unsupported file format is uploaded ( post will not be created ) 

### Issues

- fixes #3923 

### Todo

- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
Hey @dasari810 ,
sorry for not caring. I have to help somebody as an emergency who has problems with corona in his family. Because of this and other things unfortunately I need til Monday to find time for your PR. 😓

**Comment by dasari810** [comment]
hey @Tirokk  it's completely fine  :+1:  , and also take care of your self in these difficult situations :blush:

File: webapp/components/ImageUploader/ImageUploader.spec.js [hunk 1]
```diff
@@ -27,11 +27,17 @@ describe('ImageUploader.vue', () => {
       beforeEach(() => jest.useFakeTimers())
       const message = 'File upload failed'
       const fileError = { status: 'error' }
+      const unSupportedFileMessage =
+        'Please upload an image of file format : jpg , jpeg , png or gif'
```
Comment (Tirokk) [inline webapp/components/ImageUploader/ImageUploader.spec.js:31]: If you add a translation, then check for its identifier here …

File: webapp/components/ImageUploader/ImageUploader.spec.js [hunk 2]
```diff
@@ -27,11 +27,17 @@ describe('ImageUploader.vue', () => {
...
 
       it('shows an error toaster when verror is called', () => {
         wrapper.vm.onDropzoneError(fileError, message)
         expect(mocks.$toast.error).toHaveBeenCalledWith(fileError.status, message)
       })
+      it('shows an error toaster when unSupported file is uploaded', () => {
```
Comment (Tirokk) [inline webapp/components/ImageUploader/ImageUploader.spec.js:37]: Cool that you implemented a test for the toast message! 🤗😎
@dasari810 👍🏼

File: webapp/components/ImageUploader/ImageUploader.vue [hunk 3]
```diff
@@ -73,7 +74,20 @@ export default {
...
+    onUnSupportedFormat(status, message) {
+      this.$toast.error(status, message)
+    },
     initCropper(file) {
+      const supportedFormats = ['image/jpg', 'image/jpeg', 'image/png', 'image/gif']
+
+      if (supportedFormats.indexOf(file.type) < 0) {
+        this.onUnSupportedFormat(
+          'error',
+          'Please upload an image of file format : jpg , jpeg , png or gif',
```
Comment (Tirokk) [inline webapp/components/ImageUploader/ImageUploader.vue:87]: Please lets use locales for translations here @dasari810 🙏🏼
That the users have their languages.

Would be nice if you can write the possible file formats as capitals like in your other PR #3925 . 🤗

**Review by Tirokk — CHANGES_REQUESTED** [review]
Hey @dasari810 ,
nice that you did this PR as well. 🚀💫

This makes thinks much more clear and brings our project forward !!!

I have little changes requested …

**Review by Tirokk — APPROVED** [review]
Hey cool @dasari810 ,
thanks for contributing again !!! 🚀💫

A nice next step for a better user experience!

PS: I allowed my self to add the German language and add the branch `3921-set-up-travis-builds` so that the Travis builds can run now …

PPS: In case you are keen on being perfect:
There are two toast messages if an error occurs. But I already approved your PR … 🤗

Bildschirmfoto 2020-11-03 um 15 15 29

**Comment by dasari810** [comment]
yeah @Tirokk  , i will try to resolve it .

**Comment by Tirokk** [comment]
Hey cool! 👍🏼😎😁

PS: I merged new master with Travis working now in your branch

**Comment by Tirokk** [comment]
Shall I merge it @dasari810 and we create an additional issue?

**Comment by Mogge** [comment]
@dasari810 I merge this PR. Thank you very much for your contribution. I am looking forward to see you again!

**Pull request merged**